packages/faceted-search/src/components/Badges/BadgeTags/BadgeTagsForm.component.js

```diff
@@ -0,0 +1,184 @@
+	};
+	return (
+		<React.Fragment>
+			<Checkbox
+				onChange={onChangeTag}
+				aria-describedby={describedby}
+				id={`${id}-checkbox`}
+				label={label}
+				checked={checked}
+			/>
+			<div id={describedby} className="sr-only">
+				{label}
+			</div>
+		</React.Fragment>
+	);
+};
+
+BadgeTag.propTypes = {
+	id: PropTypes.string.isRequired,
+	label: PropTypes.string.isRequired,
+	onChange: PropTypes.func.isRequired,
+	checked: PropTypes.bool,
+};
+
+const createTagEntity = value => checkbox => {
+	const entity = value.find(v => v.id === checkbox.id);
+	return {
+		id: checkbox.id,
+		label: checkbox.label,
+		checked: entity ? entity.checked : checkbox.checked || false,
+	};
+};
+
+const getTags = (tagsValues, value) => {
+	return tagsValues.map(createTagEntity(value));
+};
+
+const getVisibleTags = (tags, filterValue, showAll) => {
+	const formatFilterValue = filterValue.trim().toLocaleLowerCase();
+
+	return tags
+		.filter(checkbox => get(checkbox, 'label', '').toLocaleLowerCase().includes(formatFilterValue))
+		.filter(tag => (showAll ? true : tag.checked));
+};
+
+const BadgeTagsForm = ({ tagsValues, id, onChange, onSubmit, value, feature, isLoading, t }) => {
+	const [filter, setFilter] = useState('');
+	const [showAll, setShowAll] = useState(true);
+
+	const badgeTagsFormId = `${id}-tags-form`;
+
+	const tags = useCallback(getTags(tagsValues, value, filter), [tagsValues, value]);
+	const visibleTags = useCallback(getVisibleTags(tags, filter, showAll), [tags, filter, showAll]);
+
+	const applyDataFeature = useMemo(() => getApplyDataFeature(feature), [feature]);
+
+	const onChangeTags = (event, checkboxId) => {
+		const entity = tags.find(checkboxValue => checkboxValue.id === checkboxId);
+		if (entity) {
+			entity.checked = event.target.checked;
+		}
+		onChange(
+			event,
+			tags.filter(c => c.checked),
+		);
+	};
+
+	const leftBtnLabel = showAll
+		? t('SELECTED_TAGS', { count: value.length, defaultValue: '{{count}} selected' })
+		: t('SHOW_ALL_TAGS', { defaultValue: 'Show all' });
+
+	return (
+		<React.Fragment>
+			<FilterBar
+				autoFocus={false}
+				dockable={false}
+				docked={false}
+				iconAlwaysVisible
+				id={`${badgeTagsFormId}-filter`}
+				placeholder={t('FIND_TAG_FILTER_PLACEHOLDER', {
+					defaultValue: 'Find a tag',
+				})}
+				onToggle={() => setFilter('')}
+				onFilter={(_, filterValue) => setFilter(filterValue)}
+				value={filter}
+				disabled={isLoading}
+			/>
+			{isLoading && (
+				<div className={theme('fs-badge-tags-form-loading')}>
+					<Loader size={CIRCULAR_PROGRESS_SIZE.default} />
+				</div>
+			)}
+			{!isLoading && (
+				<form
+					className={theme('fs-badge-tags-form')}
+					id={`${badgeTagsFormId}-form`}
+					onSubmit={onSubmit}
+				>
+					<RichLayout.Body id={badgeTagsFormId} className={theme('fs-badge-tags-form-body')}>
+						{!visibleTags.length && (
+							<span className={theme('fs-badge-tags-form-empty')}>
+								{t('TAG_FACET_NO_RESULT', {
```

Comment:
```suggestion
								{t('FIND_TAG_NO_RESULT', {
```
Suggestion to stay consistent with the previous key `FIND_TAG_FILTER_PLACEHOLDER`.